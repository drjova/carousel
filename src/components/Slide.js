import React from 'react';
import PropTypes from 'prop-types';

const Slider = ({ image, width, height }) => {
  const styles = {
    backgroundImage: `url(${image})`,
    backgroundPosition: 'center top',
    backgroundRepeat: 'no-repeat',
    width: `${width}px`,
    height: `${height}px`,
  };
  return <div className="slide" style={styles} />;
};

Slider.defaultProps = {
  width: 300,
  height: 600,
};

Slider.propTypes = {
  image: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
};

export default Slider;
