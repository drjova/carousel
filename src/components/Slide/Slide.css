.slide img {
  width: 300px;
  height: 300px;
}
.slide h3 {
  color: #4a739e;
}
