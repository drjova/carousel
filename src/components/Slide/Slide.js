import React from 'react';
import PropTypes from 'prop-types';

import './Slide.css';

const Slide = ({ image }) => {
  return (
    <div className="slide">
      <img src={image} />
      <h3>Title</h3>
    </div>
  );
};

Slide.propTypes = {
  image: PropTypes.string.isRequired,
};

export default Slide;
