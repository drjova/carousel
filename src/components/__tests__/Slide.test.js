import React from 'react';
import { mount } from 'enzyme';

import Slide from './../Slide';

describe('Slide component', () => {
  it('should render ``slide``', () => {
    const component = mount(<Slide image={'url'} width={300} heigh={300} />);
    expect(component).toMatchSnapshot();
  });
});
