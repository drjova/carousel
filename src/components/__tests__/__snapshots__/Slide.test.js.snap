// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component should render \`\`slide\`\` 1`] = `
<div
  className="slide"
>
  <img
    src="url"
  />
  <h3>
    Title
  </h3>
</div>
`;
